language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/foldingrange/FoldingRangeFinder.java

```diff
@@ -61,10 +62,19 @@ public void visit(RecordTypeDescriptorNode recordTypeDescriptorNode) {
     }
 
     public void visit(IfElseStatementNode ifElseStatementNode) {
-        LineRange lineRange = ifElseStatementNode.lineRange();
+        LineRange ifElseLineRange = ifElseStatementNode.lineRange();
+        LineRange ifBodyLineRange = ifElseStatementNode.ifBody().lineRange();
+        foldingRanges.removeIf(foldingRange -> ifElseLineRange.startLine().line() == foldingRange.getStartLine());
+        foldingRanges.add(createFoldingRange(ifElseLineRange.startLine().line(), ifBodyLineRange.endLine().line() - 1,
+                ifElseLineRange.startLine().offset(), ifBodyLineRange.endLine().offset(), REGION));
```

Comment:
This is a wrong usage of the kind. You should use `FoldingRangeKind` enum.

Reply:
Updated with https://github.com/ballerina-platform/ballerina-lang/pull/31015/commits/e15e37045531b60761a0460330345d91078c9bcb